tests/testing-framework/src/runtimes/spin_containerd_shim.rs

```diff
@@ -0,0 +1,229 @@
+use std::{
+    path::Path,
+    process::{Command, Stdio},
+};
+
+use crate::{
+    http::{Request, Response},
+    io::OutputStream,
+    runtimes::spin_cli::{kill_process, IoMode, SpinConfig},
+    Runtime, TestEnvironment,
+};
+
+use anyhow::Context as _;
+
+pub struct SpinShim {
+    process: std::process::Child,
+    #[allow(dead_code)]
+    stdout: OutputStream,
+    stderr: OutputStream,
+    io_mode: IoMode,
+}
+
+impl SpinShim {
+    pub fn regisry_push<R>(
+        spin_binary_path: &Path,
+        image: &str,
+        env: &mut TestEnvironment<R>,
+    ) -> anyhow::Result<()> {
+        // TODO: consider enabling configuring a port
+        Command::new(spin_binary_path)
+            .args(["registry", "push"])
+            .arg(image)
+            .current_dir(env.path())
+            .output()
+            .context("failed to push spin app to registry with 'spin'")?;
+        // TODO: assess output
+        Ok(())
+    }
+
+    pub fn image_pull(ctr_binary_path: &Path, image: &str) -> anyhow::Result<()> {
+        // TODO: consider enabling configuring a port
+        Command::new(ctr_binary_path)
+            .args(["image", "pull"])
+            .arg(image)
+            .output()
+            .context("failed to pull spin app with 'ctr'")?;
+        // TODO: assess output
+        Ok(())
+    }
+
+    /// Start the Spin app using `ctr run`
+    /// Equivalent of `sudo ctr run --rm --net-host --runtime io.containerd.spin.v2 ttl.sh/myapp:48h ctr-run-id bogus-arg` for image `ttl.sh/myapp:48h` and run id `ctr-run-id`
+    pub fn start<R>(
+        ctr_binary_path: &Path,
+        env: &mut TestEnvironment<R>,
+        image: &str,
+        ctr_run_id: &str,
+    ) -> anyhow::Result<Self> {
+        let port = 80;
+        let mut ctr_cmd = std::process::Command::new(ctr_binary_path);
+        let child = ctr_cmd
+            .arg("run")
+            .args(["--rm", "--net-host", "--runtime", "io.containerd.spin.v2"])
```

Comment:
I seem to require setting `--memory`. When I run `ctr run --help` it shows the default for `--memory` to be `0` 🤷